src/manifold/src/edge_op.cpp

```diff
@@ -152,100 +152,115 @@ void Manifold::Impl::SimplifyTopology() {
 
   // In the case of a very bad triangulation, it is possible to create pinched
   // verts. They must be removed before edge collapse.
-  SplitPinchedVerts();
+  if ( ManifoldParams().splitPinchedVerts )
+  {
+    SplitPinchedVerts();
+  }
 
-  Vec<SortEntry> entries;
-  entries.reserve(nbEdges / 2);
-  for (int i = 0; i < nbEdges; ++i) {
-    if (halfedge_[i].IsForward()) {
-      entries.push_back({halfedge_[i].startVert, halfedge_[i].endVert, i});
+  if ( ManifoldParams().splitDuplicateEdges )
+  {
+    Vec<SortEntry> entries;
+    entries.reserve(nbEdges / 2);
+    for (int i = 0; i < nbEdges; ++i) {
+      if (halfedge_[i].IsForward()) {
+        entries.push_back({halfedge_[i].startVert, halfedge_[i].endVert, i});
+      }
```

Comment:
splitPinchedVerts and splitDuplicateEdges are not optional - they rarely make a change, but when they do, it's necessary for manifoldness.

Let's just put an early return after this section when `ManifoldParams().cleanupTriangles = false` and have that be the only new param.